bouter/embedded/__init__.py

```diff
@@ -1,25 +1,45 @@
-import utilities
-from bouter import Experiment
-import flammkuchen as fl
 import numpy as np
+import pandas as pd
+
+from bouter import utilities, decorators, bout_stats
+from bouter import Experiment
 
 
 class EmbeddedExperiment(Experiment):
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args, continue_curvature=5, **kwargs):
         super().__init__(*args, **kwargs)
 
-    def vigor(self, vigor_duration=0.05):
+        if continue_curvature:
+            utilities.fill_out_segments(
+                self.tail_points_matrix, continue_curvature=continue_curvature
+            )
+
+    @property
+    def n_tail_segments(self):
+        return self["behavior"]["tail"]["n_segments"]
+
+    @property
+    def tail_points_matrix(self):
+        """Return matrix with the tail points.
+        Careful, the array is not copied!
+        """
+        columns = [f"theta_{i:02}" for i in range(self.n_tail_segments)]
```

Comment:
nice use of fstrings, I din't know you can combine them with formatting that easily!